(DOCSP-10539): require() for playgrounds (#17)
mongodb/docs-mongodb-vscode: source/require-modules.txt

.. _vsce-require:

=============================================
Use ``require()`` to Include External Modules
=============================================

.. default-domain:: mongodb

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

.. include:: /includes/fact-vsce-preview.rst

.. important::

   A complete description of Node.js, modules, and the ``require()``
   function is out of scope for this tutorial. To learn more, refer to
   the `Node.js Documentation <https://nodejs.org/api/modules.html>`__.

You can use the ``require()`` function in your MongoDB Playgrounds to
include modules which exist in separate files.

Require Native Modules
----------------------

> You can ``require()`` native Node modules (such as ``fs``) in your

Reviewer: [optional suggestion] Link ``fs`` to https://nodejs.org/api/fs.html#fs_file_system? I like the seealso to explain the specific command, but I think it's a little strange without having linked to 'fs' first/ breaks the pattern below of linking to the thing you introduce.

Author: Did a bit more standardizing in the examples. Good call.

   the `Node.js Documentation <https://nodejs.org/api/modules.html>`__.

You can use the
`require() <https://nodejs.org/api/modules.html#modules_require_id>`__
function in your MongoDB Playgrounds to
include modules which exist in separate files.

Require Native Modules
----------------------

You can ``require()`` native Node modules (such as
`fs <https://nodejs.org/api/fs.html#fs_file_system>`__) in your
Playground without any additional setup or configuration.

.. example::

   The following Playground uses the ``fs`` module to write a document
   from the ``test.employees`` collection to a file named
   ``employee.txt``:

   .. code-block:: javascript

      const fs = require('fs');

      use("test");

      const document = db.employees.findOne();

      fs.writeFileSync('employee.txt', JSON.stringify(document));

   .. seealso::

      - `fs.writeFileSync
        <https://nodejs.org/api/fs.html#fs_fs_writefilesync_file_data_options>`__

      - `require()
        <https://nodejs.org/api/modules.html#modules_require_id>`__

Require Non-Native Modules
--------------------------

To ``require()`` non-native Node modules (such as those downloaded from

> `npm <https://www.npmjs.com/>`__) you must install the package to one of
> the appropriate VS Code module folders for your operating system:

Reviewer 2: ```suggestion
`npm <https://www.npmjs.com/>`__) you must install the package in one of
these folders for your operating system:
```
Some of these folders are not VS Code specific. Except for the ones that have `.vscode` in the path, they are where Node looks for global modules by default.

Author: Agree on both suggestions. I think I originally had "copy to" which is how "to" ended up in there, but now that it's "install", "in" makes more sense. Thanks for the suggestions Max. Will update and publish.
